### PR TITLE
Fix whitelisting from the query log (and DB query log)

### PR DESCRIPTION
**Pull request by Mcat12**
**By submitting this pull request, I confirm the following:** `{please fill any appropriate checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [x] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#git-commit---signoff))

---

**What does this PR aim to accomplish?:**
Fix whitelisting not working in the query log for regex blocked domains.
https://discourse.pi-hole.net/t/dev-when-whitelisting-in-query-log-then-blocked-regex-wildcard-domain-is-added-to-blacklist/12302

**How does this PR accomplish the above?:**
Add regex blocked query types to the list of query types to whitelist when the button is clicked.

**What documentation changes (if any) are needed to support this PR?:**
None

### CODE REVIEW

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/dev-when-whitelisting-in-query-log-then-blocked-regex-wildcard-domain-is-added-to-blacklist/12302/6

File: scripts/pi-hole/js/db_queries.js [hunk 1]
```diff
@@ -273,7 +273,7 @@ $(document).ready(function() {
     });
     $("#all-queries tbody").on( "click", "button", function () {
         var data = tableApi.row( $(this).parents("tr") ).data();
-        if (data[4] === "1")
+        if (data[4] === 1 || data[4] === 4 || data[5] === 5)
```
Comment (DL6ER) [inline scripts/pi-hole/js/db_queries.js]: I cannot test this right now, but you change from comparing to a string to comparing to a number. This is important sue to the `===` operator. Is that change intended? If so, did it actually work *at all* before your change?

Comment (Mcat12) [reply scripts/pi-hole/js/db_queries.js]: This change is intended, and it was broken previously.
As seen in the `switch(data[4])` block, `data[4]` is expected to be a number instead of a string.
